src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetsControllerTests.cs

```diff
@@ -43,27 +43,56 @@ public class ListDataSetsTests(
         [Fact]
         public async Task PublicationHasSingleDataSet_Success_CorrectViewModel()
         {
-            Publication publication = DataFixture.DefaultPublication();
+            Publication publication = DataFixture
+                .DefaultPublication()
+                .WithReleases(
+                    DataFixture
+                        .DefaultRelease(publishedVersions: 1, draftVersion: true)
+                        .Generate(3)
+                );
+
+            var liveReleaseVersion = publication.ReleaseVersions.First(rv => rv.Published is not null);
```

Comment:
Does this test rely on the there being a total of three releases?

I was struggling to understand this but I think this code is picking any 1 of the 3 published release versions, and then getting a draft version from one of the two _other_ releases.

I'm wondering if it would help by creating variables for the releases prior to declaring the publication?

```csharp
var (release1, release2) = _fixture.DefaultRelease(publishedVersions: 1, draftVersion: true).Generate(2).ToTuple2();
var liveReleaseVersion = release1.Versions.Single(rv => rv.Published is not null);
var draftReleaseVersion = release2.Versions.Single(rv => rv.Published is null);
```

or something like this?

Reply:
I just wanted to prove that we'd end up with 3 `previousReleaseId` entries in the view model, rather than just the 2 that are attached to the live version and the draft version, should anyone be tempted to switch it up in the future and accidentally break this scenario.  I've added a comment as to why we're generating 3 here.